drivers/ieee802154 Kconfig and build cleanup

tbursztyka opened the pull request.

Commit:
drivers/ieee802154: Cleanup Kconfig

A very old reference to former net stack was still lurking around.
Removing it.

Taking the opportunity to clear up dependencies.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
drivers/ieee802154: Cleaning up CMakeLists.txt

Old reference on driver's specific raw mode was still lurking around.
Removing those.

Fixes #5270

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

SebastianBoe commented:
Should this reference to RAW also be removed?

```
/home/sebo/zephyr/boards/arm/usb_kw24d512/Kconfig.defconfig
109:if IEEE802154_MCR20A || IEEE802154_MCR20A_RAW
132:endif # IEEE802154_MCR20A || IEEE802154_MCR20A_RAW
```

SebastianBoe commented:
Removing myself as reviewer because I still haven't figured out how Kconfig should be written.

Commit:
boards/usb_kw24d512: Cleaning up defconfig

Old reference on driver's specific raw mode was still lurking around.
Removing those.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

tbursztyka commented:
/boards/arm/usb_kw24d512/Kconfig.defconfig  indeed

carlescufi commented:
This PR is required for the upcoming OpenThread one.

SebastianBoe commented on drivers/ieee802154/Kconfig:
```diff
@@ -9,11 +9,15 @@
+menuconfig IEEE802154
+	bool "IEEE 802.15.4 drivers options"
+	default n
+	default y if NET_L2_IEEE802154
```
Funny, just did the same change ...

I'll merge this into the OpenThread branch ...

nashif commented:
recheck

carlescufi commented:
@SebastianBoe are you OK with this? looks good to me

SebastianBoe commented:
LGTM
